Sources/SentrySwiftUI/TracingView.swift

```diff
@@ -0,0 +1,58 @@
+#if canImport(SwiftUI) && canImport(UIKit) && os(iOS) || os(tvOS)
+
+import Foundation
+import SwiftUI
+
+#if CARTHAGE || SWIFT_PACKAGE
+@_implementationOnly import SentryInternal
+#endif
+
+@available(iOS 13, macOS 10.15, tvOS 13, *)
+struct TracingView: UIViewRepresentable {
+    
+    let name: String
+    let waitForFullDisplay: Bool
+    let tracer: SentryTracer?
+    
+    class SentryTracingView: UIView {
+        
+        let tracer: SentryTracer?
+        let tracker: SentryTimeToDisplayTracker?
+        
+        init(name: String, waitForFullDisplay: Bool, tracer: SentryTracer?) {
+            self.tracer = tracer
+            if let tracer = self.tracer {
+                let tracker = SentryTimeToDisplayTracker(name: name, waitForFullDisplay: waitForFullDisplay)
+                self.tracker = tracker
+                SentryUIViewControllerPerformanceTracker.shared.setTimeToDisplay(tracker)
+                tracker.start(for: tracer)
+            } else {
+                tracker = nil
+            }
+            
+            super.init(frame: CGRect(origin: .zero, size: CGSize(width: 1, height: 1)))
+        }
+        
+        required init?(coder: NSCoder) {
+            fatalError("init(coder:) has not been implemented")
+        }
+      
+        override func didMoveToWindow() {
```

Comment:
`didMoveToSuperView` can happen outside the view hierarchy.
We actually dont care about the parent, we just need to know that the view was added to the view hierarchy and is about to be rendered.

Reply:
Can you please further elaborate how we know that `didMoveToSuperView` is for sure called when the view is rendered? On iOS it is possible to have multiple windows, even on iOS (I have done that before), calling `UIWindow.makeKeyAndVisible()` for the main window, while others are stacked by deriving instances from a `UIWindowScene` and setting `isHidden` to `false`.

In most cases this will probably be good enough, just want to point out that there might be unexpected edge cases.

Reply:
Ah yes, I ment to write didMoveToWindow. Thanks for elaborating